### PR TITLE
fix: handle UNAPPROVED_ABOVE status

### PR DESCRIPTION
**Pull request by mediremi**
Handle the `UNAPPROVED_ABOVE` status in the same way we handle `UNAPPROVED_WAITING` and `UNAPPROVED_ELSEWHERE`.

### CODE REVIEW

File: src/shared/status-tag/use-approval-state.js
Comment (Mohammer5) [inline src/shared/status-tag/use-approval-state.js:40]: If I get [these docs](https://docs.dhis2.org/en/develop/developing-with-the-android-sdk/workflow.html) (search for "UNAPPROVED_ABOVE") right, the this means that it's already been approved and the next level has to approve it, but I'm really not sure if I'm interpreting the docs correctly

**Review by Mohammer5 — CHANGES_REQUESTED** [review]

File: src/shared/status-tag/use-approval-state.js
Comment (Mohammer5) [reply src/shared/status-tag/use-approval-state.js:40]: What I mean is: Is "waiting" the right label?

Comment (HendrikThePendric) [reply src/shared/status-tag/use-approval-state.js:40]: Yeah, I’m not sure about the correct interpretation either, but we need to get these confirmed anyway, so I guess we can add this state to this case (waiting).

Now that we already encountered this undocumented state on day 1 already, I think we should:
- Definitely talk to Lars/Jim about the correct interpretation of these states
- Probably scan the dhis-core source code for a complete overview of the statuses

Probably it makes sense to ensure we have a complete list before we talk to one of them…

Comment (HendrikThePendric) [reply src/shared/status-tag/use-approval-state.js:40]: PS: I am referring to this state as undocumented because I don’t see it in the list of approval states. Could be I’m not looking properly or in the wrong place…

Comment (HendrikThePendric) [reply src/shared/status-tag/use-approval-state.js:40]: I had a look in the dhis2-core code-base and was expecting the relevant section of code to be in the [DataApprovalState.java](https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataapproval/DataApprovalState.java#L43-L100) file. But that only contained the 7 states below and we already had comprised a list of 8 [from the docs](https://docs.dhis2.org/en/full/develop/dhis-core-version-master/developer-manual.html#webapi_data_approval):
```
UNAPPROVABLE
UNAPPROVED_ABOVE
UNAPPROVED_WAITING
UNAPPROVED_READY
APPROVED_ABOVE
APPROVED_HERE
ACCEPTED_HERE
```
Then looked in [dataApproval.js](https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-web/dhis-web-approval/src/main/webapp/dhis-web-approval/javascript/dataApproval.js#L508-L611) (I think the is the Javascript part of the old struts app) and there I found 14 states:
```
ACCEPTED_ELSEWHERE
ACCEPTED_HERE
APPROVED_ABOVE
APPROVED_ELSEWHERE
APPROVED_HERE
PARTIALLY_ACCEPTED_ELSEWHERE
PARTIALLY_ACCEPTED_HERE
PARTIALLY_APPROVED_ELSEWHERE
PARTIALLY_APPROVED_HERE
UNAPPROVABLE
UNAPPROVED_ABOVE
UNAPPROVED_ELSEWHERE
UNAPPROVED_READY
UNAPPROVED_WAITING
```
What was puzzling to is how the JavaScript file could have to deal with more approval states than defined in the Java code. I think we need to get more info on this.

For now, I guess we could just add the case `UNAPPROVED_ABOVE` as you suggested, but perhaps it would be good to also add a default case which throws a descriptive error message?

**Review by Mohammer5 — APPROVED** [review]

**Pull request merged**
